src/react-components/room/TipContainer.js

```diff
@@ -86,28 +31,68 @@ export function RecordModeTip() {
 }
 
 export function TipContainer({ hide, inLobby, inRoom, isStreaming, isEmbedded, scene, store, hubId, presences }) {
-  const intl = useIntl();
   const [lobbyTipDismissed, setLobbyTipDismissed] = useState(false);
   const [broadcastTipDismissed, setBroadcastTipDismissed] = useState(() =>
     store.state.confirmedBroadcastedRooms.includes(hubId)
   );
   const [streamingTipDismissed, setStreamingTipDismissed] = useState(false);
   const [embeddedTipDismissed, setEmbeddedTipDismissed] = useState(false);
   const [onboardingTipId, setOnboardingTipId] = useState(null);
+  const timeoutRef = useRef(null);
+  const [wasHidden, setWasHidden] = useState(false);
 
   const onSkipOnboarding = useCallback(() => {
-    scene.systems.tips.skipTips();
-  }, [scene]);
+    setOnboardingTipId(null);
+    timeoutRef.current = setTimeout(() => {
+      setWasHidden(false);
+      scene.systems.tips.skipTips();
+    }, 200);
+  }, [scene, timeoutRef]);
+
+  const onNextTip = useCallback(() => {
+    setOnboardingTipId(null);
+    timeoutRef.current = setTimeout(() => {
+      setWasHidden(false);
+      scene.systems.tips.nextTip();
+    }, 200);
+  }, [scene, timeoutRef]);
+
+  const onPrevTip = useCallback(() => {
+    setOnboardingTipId(null);
+    timeoutRef.current = setTimeout(() => {
+      setWasHidden(false);
+      scene.systems.tips.prevTip();
+    }, 200);
+  }, [scene, timeoutRef]);
+
+  useEffect(() => {
+    if (isEndTooltipStep(onboardingTipId)) {
+      timeoutRef.current = setTimeout(() => {
+        scene.systems.tips.nextTip();
```

Comment:
Not sure how exactly to untangle but the control flow here feels kinda messy. I click next, then 200ms later this function in the tip system ends up setting the tip as resolved, which will cause the active tip to change in the next  tip system tick, emitting a `tip-changed` event, which gets picked up in the event listener set up in the effect hook below, setting the tip first to null, then 200ms later setting it to the tip from the event. That is qutie a roundabout way of incrementing a number :)

Not mandatory since I think this was supposed to just be a quick fix and I think you were trying to minimize changes (and this implementation seems to work), but I think this can be simplified. We have a set of tips we want to step through, via next/prev buttons. This state should be able to live entirely in react land and just increment the current step coutner. Additionally, the current step can be completed by performing the action instead of clicking next. This checking makes sense to be done in a system since it needs to interface with game logic.

The previous implementation of tips skipped tips you have already completed. Seems reasonable to drop that requirement though since this is now more of a wizard you step through than the original intent, which was to provide a constant series of tips until you happened to finish them all. You now either complete the onbaording, or skip it, and thats all we should really need to save. If its still desired it would complicate the logic a bit and maybe it ends up similar to this.

Reply:
Yes, I don't love it either. I've done some refactoring, the 200ms we part of some initial implementation but are not necessary so I've removed those. Also I had to do some state handling to manage the different scenarios when the notifications are hidden because the UI is removed (ie. fullscreen mode is entered) and we need to play the animations again or not. I've simplified that too.